src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/GenericTypeParameterBuilder.cs

```diff
@@ -119,8 +119,6 @@ public override Type MakeArrayType(int rank)
 
         [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)]
         [return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)]
-        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2063:UnrecognizedReflectionPattern",
-            Justification = "Linker doesn't recognize always throwing method. https://github.com/mono/linker/issues/2025")]
```

Comment:
We have references to this bug under src/coreclr/nativeaot as well - how was the redundant suppression analysis for NativeAOT done? I wonder why it didn't find those (they should be redundant because they were just a bug workaround).

Reply:
The redundant warning suppression detection has not been ported to nativeaot yet (#75201). The suppression you mentioned should be identified after the functionality is ported and run witihin nativeaot.

Reply:
This PR has a change in src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/Runtime/General/Assignability.cs - what mechanism was used to detect it?

(NativeAOT doesn't run on code that is unreferenced and doesn't have a "library mode" like illink does - so the ability to detect redundant suppressions is somewhat limited even if we were to port the feature - we would have to come up with an app that includes all the code in NativeAOT's libraries. It will only flag things that are used, making it easy for us to ship with an actual warning that only some customers will see - customers who happen to use that part of the library.)

Reply:
It was detected because linker was used to "trim libraries for warnings" (The `sfx` project run of linker) and it was given the AOT corelib as input. Jeremi found out that this happens when you run `build Clr.Aot+Libs -rc Debug -lc Release`, in which case linker gets `artifacts\bin\coreclr\windows.x64.Debug\aotsdk\System.Private.CoreLib.dll` as one of the input assemblies.

As for the comment about NativeAOT not having libraries mode - that should not matter. The detection works in the "trim app" mode as well - it only reports redundant warnings on methods which were "marked".

Reply:
> It was detected because linker was used to "trim libraries for warnings" (The sfx project run of linker) and it was given the AOT corelib as input

I see, yes, there can only be one corelib project for the libs partition but we have 3 in this repo.

The corelib is selected here: https://github.com/dotnet/runtime/blob/e8b149109c8b04161b83d0e4c997b63759c6d936/Directory.Build.props#L261-L263